code/game/dna/dna_modifier.dm

```diff
@@ -239,35 +240,35 @@
 	else if(M.anchored)
 		return FALSE
 
-	if(ismanifested(M) || !iscarbon(M))
-		if(user)
-			to_chat(user, "<span class='notice'>For some reason, the scanner is unable to read that person's genes.</span>")
+	if(ismanifested(M) || (!iscarbon(M) && !istype(M, /mob/living/slime_pile) && !istype(M, /mob/living/gore_pile)))
+		if(usr)
```

Comment:
user was undefined and it was a longstanding bug in the cloner code where checks like if(user) would always return false.
usr is the hardcoded thing akin to src as I understand.

Reply:
`usr` is hardcoded to be "the last player to have done a thing (in a context related to `src`)". You have no control over it. It's hard to see the code flow and debug things with `usr`. DM's own reference recommends against using `usr` in procs that are not natively called by BYOND, such as `Topic`, etc.

If `user` is null, then the bug is something further above and that should be fixed instead of relying on `usr`.